### PR TITLE
Bump up ZK version to 3.6.3

### PR DESCRIPTION
**Pull request by jackjlli**
In LinkedIn the ZK version has already been bumped up to 3.6.3 and it's working as expected. 
This PR bumps up ZK version to 3.6.3 in the Apache Pinot repo.

Issue: https://github.com/apache/pinot/issues/9431

### CODE REVIEW

**Comment by siddharthteotia** [comment]
FYI @walterddr @Jackie-Jiang

File: pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/pom.xml [hunk 1]
```diff
@@ -107,6 +107,10 @@
       <groupId>net.sf.jopt-simple</groupId>
       <artifactId>jopt-simple</artifactId>
     </dependency>
+    <dependency>
```
Comment (walterddr) [inline pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/pom.xml:110]: Is this pom change related to zk version upgrade?

Comment (jackjlli) [reply pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/pom.xml:110]: Unfortunately yes, the metric-core module is required. Otherwise, the following exception would be thrown:
```
Error:  org.apache.pinot.plugin.stream.kafka20.KafkaPartitionLevelConsumerBackwardCompatibilityTest.setUp  Time elapsed: 0.906 s  <<< FAILURE!
java.lang.NoClassDefFoundError: com/codahale/metrics/Reservoir
	at org.apache.zookeeper.metrics.impl.DefaultMetricsProvider$DefaultMetricsContext.lambda$getSummary$2(DefaultMetricsProvider.java:126)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at org.apache.zookeeper.metrics.impl.DefaultMetricsProvider$DefaultMetricsContext.getSummary(DefaultMetricsProvider.java:122)
	at org.apache.zookeeper.server.ServerMetrics.<init>(ServerMetrics.java:74)
	at org.apache.zookeeper.server.ServerMetrics.<clinit>(ServerMetrics.java:44)
	at org.apache.zookeeper.server.persistence.FileTxnSnapLog.restore(FileTxnSnapLog.java:246)
	at org.apache.zookeeper.server.ZKDatabase.loadDataBase(ZKDatabase.java:285)
	at org.apache.zookeeper.server.ZooKeeperServer.loadData(ZooKeeperServer.java:494)
	at org.apache.zookeeper.server.ZooKeeperServer.startdata(ZooKeeperServer.java:665)
	at org.apache.zookeeper.server.NIOServerCnxnFactory.startup(NIOServerCnxnFactory.java:758)
	at org.apache.zookeeper.server.ServerCnxnFactory.startup(ServerCnxnFactory.java:130)
	at org.apache.pinot.plugin.stream.kafka20.utils.EmbeddedZooKeeper.<init>(EmbeddedZooKeeper.java:42)
	at org.apache.pinot.plugin.stream.kafka20.utils.MiniKafkaCluster.<init>(MiniKafkaCluster.java:54)
	at org.apache.pinot.plugin.stream.kafka20.KafkaPartitionLevelConsumerTest.setUp(KafkaPartitionLevelConsumerTest.java:63)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:108)
	at org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:523)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:224)
	at org.testng.internal.Invoker.invokeConfigurations(Invoker.java:146)
	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:166)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:105)
	at org.testng.TestRunner.privateRun(TestRunner.java:744)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:380)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:375)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1301)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1226)
	at org.testng.TestNG.runSuites(TestNG.java:1144)
	at org.testng.TestNG.run(TestNG.java:1115)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:136)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:112)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:99)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:145)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: java.lang.ClassNotFoundException: com.codahale.metrics.Reservoir
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 44 more
```
I can try restrict the scope of that dependency to `test` only, as I can build it locally when running tests is disabled.

Comment (jackjlli) [reply pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/pom.xml:110]: Changing it `test` scope works. It seems `ServerCnxnFactory` is only used in test classes, so we should be good with the current changes in this PR.

**Review by walterddr — APPROVED** [review]
